### DISCUSSION TITLE
How do I detect where the executable is coming from?

### DISCUSSION THREAD

**Discussion by kytta** [jdx/mise#2444] [Q&A] [answered]
I am on macOS and I want to achieve the following setup:

1. [x] `mise` is installed (shims mode)
2. [x] Python is installed
   - from Python.org (for my personal use)
   - from Homebrew (as a dependency of formulae)
   - via Xcode Command-Line Tools
3. [ ] globally, the Python from Python.org should always be used
   - so, without `mise`, `which python3` should give me the one from `Python.framework`
4. [x] if I want to, I can set a specific Python version for a project
   - in this case, `mise` takes over

I am confused about the third step. If I do not set a Python version globally, `which python3` still points me to the shims — I'd expect that. However, I do not know what the shim resolves to. So, when I run `python3`, which executable gets started?

(the reason I'm asking is that I'm not sure my `$PATH` is correctly configured, but I have, like, [5 identical versions of the latest Python](https://xkcd.com/1987/))

**Comment by nickchomey** [comment, marked as answer]
Can you use mise without shims?

**Reply by kytta** [reply]
I believe no, because I need JetBrains IDE support (I also symlink it to `~/.asdf`

**Reply by kytta** [reply]
Damn, I got confused. For some reason, I thought I have to have shims in my path for the IDEs to pick up on, but the `.asdf` symlink is enough. I've disabled shims, and `which` now shows me the correct path. Thank you!
